### PR TITLE
nixos/redis: enable vmOverCommit by defaul; don't disable transparent hugepages

### LINKED ISSUE #90602: redis warns about transparent huge pages support in kernel
**Describe the bug**

**To Reproduce**
1. Install redis via:
```
services.redis.enable = true;
services.redis.vmOverCommit = true;
```
2. Look at the `redis` log `journalctl -u redis.service` and observe:
> WARNING you have Transparent Huge Pages (THP) support enabled in your kernel. This will create latency and memory usage issues with Redis. To fix this issue run the command 'echo never > /sys/kernel/mm/transparent_hugepage/enabled' as root, and add it to your /etc/rc.local in order to retain the setting after a reboot. Redis must be restarted after THP is disabled.
3. reboot

**Expected behavior**
No warnings or errors in `redis` log.

**Additional context**
I thought [this](https://github.com/NixOS/nixpkgs/blob/a84b797b28eb104db758b5cb2b61ba8face6744b/nixos/modules/services/databases/redis.nix#L196) line was supposed to prevent such a warning... :thinking: 

**Notify maintainers**

ping @peti @flokli @Mic92 @maralorn  

**Metadata**
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.
```
 - system: `"x86_64-linux"`
 - host os: `Linux 5.4.35, NixOS, 20.03.1589.9137f05564e (Markhor)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.4`
 - channels(root): `"nixos-20.03.1589.9137f05564e"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
# a list of nixos modules affected by the problem
module: redis
```

### PR DESCRIPTION
**Pull request by SuperSandro2000**
## Description of changes

See each commits. Tested on a nixos machine by booting with transparent_hugepage=madvise and checking the redis logs which don't mention anything anymore.

Closes #90602

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by aanderse — APPROVED** [review]
this makes sense to me :+1:

**Comment by SuperSandro2000** [comment]
Just added a release note entry

**Comment by SuperSandro2000** [comment]
Do we want to merge this before the 24.05 branch off?

File: nixos/modules/services/databases/redis.nix [hunk 1]
```diff
@@ -59,7 +59,7 @@ in {
       vmOverCommit = mkEnableOption ''
         setting of vm.overcommit_memory to 1
```
Comment (flokli) [inline nixos/modules/services/databases/redis.nix:60]:
```suggestion
        set `vm.overcommit_memory` sysctl to 1
```

File: nixos/doc/manual/release-notes/rl-2405.section.md [hunk 2]
```diff
@@ -265,6 +265,8 @@ The pre-existing [services.ankisyncd](#opt-services.ankisyncd.enable) has been m
 
 - `hare` may now be cross-compiled. For that to work, however, `haredoc` needed to stop being built together with it. Thus, the latter is now its own package with the name of `haredoc`.
 
+- `services.redis.vmOverCommit` now defaults to `true` and no longer enforces Transparent Hugepages to be disabled as long as they are set to `madvise`.
```
Comment (flokli) [inline nixos/doc/manual/release-notes/rl-2405.section.md:268]: >  as long as they are set to `madvise`

I don't understand this part. This likely needs a link for more context.

Comment (SuperSandro2000) [reply nixos/doc/manual/release-notes/rl-2405.section.md:268]: madvise is the kernel default and redis only works with that. If anything else is set the following error should be triggered

https://github.com/redis/redis/blob/f95031c4733078788063de775c968b6dc85522c0/src/syscheck.c#L160-L164

https://www.kernel.org/doc/html/v4.18/admin-guide/mm/transhuge.html#:~:text=the%20near%20future.-,madvise,-will%20enter%20direct

Comment (SuperSandro2000) [reply nixos/doc/manual/release-notes/rl-2405.section.md:268]: I've reworded this section a bit.

**Pull request merged**